odoo/osv/expression.py

```diff
@@ -1260,7 +1260,7 @@ def __leaf_to_sql(self, eleaf):
             if left not in model:
                 raise ValueError("Invalid field %r in domain term %r" % (left, leaf))
             format = '%s' if need_wildcard else model._fields[left].column_format
-            unaccent = self._unaccent if sql_operator.endswith('like') else lambda x: x
+            unaccent = self._unaccent if sql_operator.endswith('like') and left != "parent_path" else lambda x: x
```

Comment:
The problem I see here is that any `fields.Char(index=True)` may see its index rendered useless by the activation of the unaccent extension. Shouldn't we instead fix the index creation so that it does `CREATE INDEX on <table> (unaccent(<field>))` instead?

`child_of` queries would benefit from this index as well.

Of course the indexed values `unaccent(parent_path)` are a bit dumb for `parent_path` specifically, but the solution works across all fields.

Reply:
You are right, I didn't realize the issue with `unaccent` was global to all char fields.
We can't create indices for `unaccent` though, because the function is not `IMMUTABLE`.

Reply:
> We can't create indices for `unaccent` though, because the function is not `IMMUTABLE`.

We could: we'd have to either make unaccent IMMUTABLE (using `ALTER FUNCTION unaccent(text) IMMUTABLE`), or wrap it with an immutable function. There are some caveats with either solutions, as discussed [here](https://stackoverflow.com/a/11007216/939860).

Considering we don't expose any way for users or admins to change dictionaries or shadow them dynamically, the first option seems safe enough in our case, and gives best performance. We've been using this technique for years and haven't encountered any issue so far.

Worst case is that some unaccent rule will change (like the collation of ligatures that changed in 9.6), and a reindex will be needed if some records falls into the new case. Not very likely, and not a big issue compared to the benefit, I think. And it can only happen for columns where non-ascii values are stored (e.g. it cannot break `parent_path`).

~

An alternative that is slightly more generic than your current patch is to support a new param `fields.Char(unaccent=False)`. With this you'd be able to turn unaccent off selectively, and make plain indexes work. That wouldn't work for fields where you want `unaccent` *and* `an index`, though.

In fact, having both options makes sense. You should be able to turn off `unaccent` selectively for columns where it makes no sense. But when `unaccent` is truly enabled on a column, `index=True` should work.

~

Either way, I'm starting to think that those changes are too much for a stable branch, and for 13.0 we should just recommend manual index creation for stable, when someone is facing a child_of performance issue with `unaccent`:
```sql
ALTER FUNCTION unaccent(text) IMMUTABLE;
CREATE INDEX on <table> (unaccent(parent_path));
```

The above changes might be better targeted to `master` - to be discussed with @rco-odoo.

Reply:
Very interesting indeed.

> Either way, I'm starting to think that those changes are too much for a stable branch

Definitely, I tried to have a very minimal patch (too minimal) for this reason :)

Should I split the `unaccent` commit from this pull request, so we can move forward with the optimization of sub-searches?

Reply:
> Should I split the `unaccent` commit from this pull request, so we can move forward with the optimization of sub-searches?

Yes please, it will certainly make merging easier :)

Reply:
Done :)